HVZ.Web/Shared/Org Authorization/OrgAuth.razor

```diff
@@ -43,20 +42,64 @@
         
         if(OrgId != null && userId != null)
         {
-            Org = await OrgRepo.GetOrgById(OrgId);
+            if (Org == null)
+                Org = await OrgRepo.GetOrgById(OrgId);
 
             IsAdmin = await OrgRepo.IsAdminOfOrg(OrgId, userId);
             IsMod = IsAdmin || await OrgRepo.IsModOfOrg(OrgId, userId);
             OwnerId = Org.OwnerId;
 
             Logger.LogDebug($"User ID: {userId}; Org ID: {OrgId}\nUser has admin: {IsAdmin}; User has mod: {IsMod}");
             shouldRender = true;
+
+            OrgRepo.AdminsUpdated += delegate(object? sender, OrgUpdatedEventArgs eventArgs)
+            {
+                Organization eventOrg = eventArgs.Org;
+
+                if (eventOrg.Id == OrgId)
+                {
+                    if (UserPermissionsChanged(eventOrg, userId))
+                    {
+                        Org = eventOrg;
+                        Refresh();
+                    }
+                }
+            };
+
+            OrgRepo.ModsUpdated += delegate(object? sender, OrgUpdatedEventArgs eventArgs)
+            {
+                Organization eventOrg = eventArgs.Org;
+
+                if (eventOrg.Id == OrgId)
+                {
+                    if (UserPermissionsChanged(eventOrg, userId))
+                    {
+                        Org = eventOrg;
+                        Refresh();
+                    }
+                }
+            };
         }
     }
 
     public void Refresh()
     {
-        StateHasChanged();
-        InvokeAsync(OnParametersSetAsync);
+        shouldRender = true;
+        IsAdmin = Org.Administrators.Contains(userId!);
+        IsMod = IsAdmin || Org.Moderators.Contains(userId!);
```

Comment:
Include a null check in Refresh() instead of null suppression

Reply:
in this case, it does not matter if `userId` is null because `HashSet.Contains(null!)` will return false